sympy/solvers/solveset.py

```diff
@@ -2510,8 +2510,8 @@ def linear_coeffs(eq, *syms, dict=False, strict=True, first=True):
         raise ValueError('duplicate symbols given')
     try:
         c, d = _lin_eq2dict(eq, symset, strict)
-    except PolyNonlinearError as exc:
-        raise NonlinearError(str(exc))
+    except PolyNonlinearError as err:
+        raise NonlinearError(str(err)) from err
```

Comment:
It would probably be good for the error message here to mention the strict flag so that users are made aware of it when they need it.